.github/workflows/cd-pipeline.yaml

```
@@ -138,24 +152,55 @@ jobs:
           # HACK: Obtain chart tarball out of the latest commit
           chart_version="$(yq ".version" charts/bitnami/${chart}/Chart.yaml)"
           aws s3 cp s3://${{ secrets.AWS_S3_BUCKET }}/bitnami/${chart}-${chart_version}.tgz download/
-          # Rebuild index
-          helm repo index --url https://charts.bitnami.com/bitnami --merge index/bitnami/index.yaml download
-          cp download/index.yaml index/bitnami/index.yaml
+
           cd index
-          # Push changes
           git config user.name "Bitnami Containers"
           git config user.email "bitnami-bot@vmware.com"
-          git add bitnami/index.yaml && git commit -m "${chart}-${chart_version}: Update index.yaml" -s && git push
+
+          attempts=0
+          max_attempts=5
+          is_index_updated=0
+          while [[ $attempts -lt $max_attempts ]]; do
+            attempts=$((attempts + 1))
+            git fetch origin index
+            git reset --hard origin/index
+
+            # Rebuild index
+            helm repo index --url https://charts.bitnami.com/bitnami --merge bitnami/index.yaml ../download
+            cp ../download/index.yaml bitnami/index.yaml
+            # Push changes
+            git add bitnami/index.yaml && git commit -m "${chart}-${chart_version}: Update index.yaml" -s
+            git push && is_index_updated=1 || true
+
+            if [[ $is_index_updated -eq 1 ]]; then
+              echo "::set-output name=result::ok"
+              break
+            else
+              echo "Failed to push during attempt $attempts"
+            fi
```

Will break inside the `if` finish the execution? I see you are unconditionally printing the fail message below.
What about this:

```suggestion
            git push && is_index_updated=1 || echo "Failed to push during attempt $attempts"
```

Then add the `$is_index_updated -eq 1` condition to the while?

---

Yes, you are right. There was some strange logic in here. I have applied your suggestions now, thanks!